x-pack/solutions/observability/plugins/observability_agent/server/tools/get_change_points/get_log_change_points.ts

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AggregationsAggregationContainer } from '@elastic/elasticsearch/lib/api/types';
+import type { IScopedClusterClient } from '@kbn/core/server';
+import type { QueryDslQueryContainer } from '@kbn/data-views-plugin/common/types';
+import type { ChangePointType } from '@kbn/es-types/src';
+import { omit, orderBy } from 'lodash';
+import { z } from '@kbn/zod';
+import { ToolType } from '@kbn/onechat-common';
+import { ToolResultType } from '@kbn/onechat-common/tools/tool_result';
+import type { BuiltinToolDefinition } from '@kbn/onechat-server';
+import type { CoreSetup, Logger } from '@kbn/core/server';
+import { SavedObjectsClient } from '@kbn/core/server';
+import type {
+  ObservabilityAgentPluginSetupDependencies,
+  ObservabilityAgentPluginStart,
+  ObservabilityAgentPluginStartDependencies,
+} from '../../types';
+import { getFilters, dateHistogram } from './common';
+
+export const OBSERVABILITY_GET_LOG_CHANGE_POINTS_TOOL_ID = 'observability.get_log_change_points';
+
+interface DateHistogramBucket {
+  key: string | number;
+  key_as_string?: string;
+  doc_count: number;
+}
+
+interface ChangePointResult {
+  type: {
+    indeterminable?: boolean;
+    [key: string]: any;
+  };
+  bucket?: {
+    key: string | number;
+    key_as_string?: string;
+  };
+}
+
+interface CategorizeTextBucket {
+  key: string;
+  doc_count: number;
+  regex: string;
+  over_time: {
+    buckets: DateHistogramBucket[];
+  };
+  changes: ChangePointResult;
+}
+
+interface SamplerAggregation {
+  groups?: {
+    buckets: CategorizeTextBucket[];
+  };
+}
+
+interface LogChangesAggregations {
+  sampler?: SamplerAggregation;
+}
+
+interface LogChangePoint {
+  time: string;
+  type: string;
+  [key: string]: any;
+}
+
+interface LogChangeResult {
+  key: string;
+  pattern: string;
+  over_time: Array<{
+    x: number;
+    y: number;
+  }>;
+  changes: { type: ChangePointType } | LogChangePoint;
+}
+
+export async function getLogChangePoint({
+  index,
+  filters,
+  field,
+  esClient,
+}: {
+  index: string;
+  filters: QueryDslQueryContainer[];
+  field: string;
+  esClient: IScopedClusterClient;
+}): Promise<LogChangeResult[]> {
+  const countDocumentsResponse = await esClient.asCurrentUser.search({
+    size: 0,
+    track_total_hits: true,
+    index,
+    query: {
+      bool: {
+        filter: filters,
+      },
+    },
+  });
+
+  const totalHits =
+    typeof countDocumentsResponse.hits.total === 'number'
+      ? countDocumentsResponse.hits.total
+      : countDocumentsResponse.hits.total?.value ?? 0;
```

Comment:
Use the util that @viduni94 added

https://github.com/elastic/kibana/blob/5a797529c91408565f3495f9f1e55cc3cc0f53a4/x-pack/solutions/observability/plugins/observability_agent/server/utils/get_hits_total.ts#L10-L13